src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -247,6 +247,8 @@ CreateInsertSelectRouterPlan(Query *originalQuery,
 	int shardCount = targetCacheEntry->shardIntervalArrayLength;
 	bool allReferenceTables = restrictionContext->allReferenceTables;
 
+	EnsureSchemaNode();
```

Comment:
Or, wait let's move it to `ErrorIfInsertSelectQueryNotSupported()` function. That should be more intuitive.